Zone5/Views/ThirdPartyConnectionsView.swift

```diff
@@ -40,21 +48,44 @@ public class ThirdPartyConnectionsView: APIView {
 		return delete(endpoint, with: completion)
 	}
 
+    /// Initialize a connection for the current user for the given 3rd party type
+    /// - Parameters
+
+    @discardableResult
+    public func initializeThirdPartyConnection(type: UserConnectionType, completion: @escaping Zone5.ResultHandler<ThirdPartyInitializeResponse>) -> PendingRequest? {
+        let endpoint = ThirdPartyEndpoint(uri: Endpoints.initializePairing.rawValue + type.connectionName)
+        return post(endpoint, body: EmptyBody(), with: completion)
+    }
+
 	/// Set an access token for the current user for the given 3rd party type
 	/// - Parameters
-	
+
 	@discardableResult
-	public func setThirdPartyToken(type: UserConnectionType, connection: ThirdPartyToken, completion: @escaping Zone5.ResultHandler<ThirdPartyResponse>) -> PendingRequest? {
-		return post(Endpoints.setThirdPartyConnection, parameters: queryParams(type), body: connection, with: completion)
+    public func setThirdPartyToken(type: UserConnectionType, parameters: URLEncodedBody?, completion: @escaping Zone5.ResultHandler<Zone5.VoidReply>) -> PendingRequest? {
+        let endpoint = ThirdPartyEndpoint(uri: Endpoints.files.rawValue + type.connectionName + "/confirm")
+        return get(endpoint, parameters: parameters, expectedType: Zone5.VoidReply.self, with: completion)
```

Comment:
Take a look at other Endpoints in other views where urls requires string replacement formatting. The pattern used is to define the Endpoints url with a variable, e.g. for confirm it would be 
`case confirm = "/rest/files/{type}/confirm"` 
and call it here in the setThirdPartyToken func with Endpoints.confirm.replacingTokens(["type": type])

Do this with all the endpoints, it will clean this file up considerably, you won't the` private class ThirdPartyEndpoint: InternalRequestEndpoint` class and each Endpoints enum will map to a distinct server endpoint, which is what it is supposed to represent, rather than it just being a static string fragment.

Reply:
Done